x/programs/rust/examples/multisig/src/lib.rs

```diff
@@ -0,0 +1,495 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use wasmlanche_sdk::{
+    public, state_keys, Address, Context, DeferDeserialize, ExternalCallError, Program,
+};
+
+pub const MIN_VOTES: u64 = 2;
+
+#[derive(Debug, BorshSerialize, BorshDeserialize)]
+pub struct Proposal {
+    to: Program,
+    method: String,
+    args: Vec<u8>,
+    value: u64,
+    executed: bool,
+    voters_len: u64,
+}
+
+#[derive(Debug, BorshSerialize, BorshDeserialize)]
+pub struct Voter {
+    address: Address,
+    voted: bool,
+}
+
+#[derive(Debug, BorshSerialize, BorshDeserialize)]
+pub enum ProposalError {
+    AlreadyExecuted,
+    AlreadyVoted,
+    InexistentProposal,
+    QuorumNotReached,
+    NotVoter,
+    ExecutionFailed(ExternalCallError),
+}
+
+#[state_keys]
+pub enum StateKeys {
+    LastProposalId,
```

Comment:
(this key was renamed to `NextProposalId` instead)
I'm not sure if we could have such a host function, the data may not be stored in a contiguous space of memory and it may be hard to have such an information. What we may want though is a `Vec` library program (if we are able to write stateful libraries in the future) that stores this information. This is a good remark !

Reply:
The reason we might want this `Vec` library is that storing a `Vec` as a state key is going to make it costly to pull it from storage because we can only get the full vec.

Reply:
yea, a wayyyys back the initial thought was to have host functions that would reduce the amount of data pulled into WASM(stuff like getting values in a map, vec, etc...). we don't have host functions for maps anymore because of `StateKeys` but vecs are a little different. 

under the hood a `Vecs` could be stored/tracked differently, and come with additional get functions like `get_index`. Could also do something similar to [solidity](https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html#mappings-and-dynamic-arrays), and just handle them i differently under the hood in get/set methods.

Reply:
That's very interesting! I don't know if we really want this at the vm level though, it feels more like a program-level thing, the vm may not need to be aware of that, but I'm happy to try both !!